crates/astria-grpc-mock-test/tests/health/main.rs

```diff
@@ -0,0 +1,270 @@
+// allow just make the tests work for now
+#![allow(clippy::should_panic_without_expect)]
+
+use std::{
+    net::SocketAddr,
+    pin::Pin,
+    sync::Arc,
+};
+
+use astria_grpc_mock::{
+    matcher,
+    response,
+    Mock,
+};
+use astria_grpc_mock_test::health::{
+    health_client::HealthClient,
+    health_server::{
+        Health,
+        HealthServer,
+    },
+    HealthCheckRequest,
+    HealthCheckResponse,
+};
+use tokio::{
+    join,
+    task::JoinHandle,
+};
+use tokio_stream::{
+    wrappers::TcpListenerStream,
+    Stream,
+};
+use tonic::{
+    transport::Server,
+    Request,
+    Response,
+    Status,
+};
+
+struct MockServer {
+    _server: JoinHandle<()>,
+    local_addr: SocketAddr,
+    mocked: astria_grpc_mock::MockServer,
+}
+
+async fn start_mock_server() -> MockServer {
+    let listener = tokio::net::TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let local_addr = listener.local_addr().unwrap();
+    let mock_server = astria_grpc_mock::MockServer::new();
+    let server = tokio::spawn({
+        let mock_server = mock_server.clone();
+        async move {
+            let _ = Server::builder()
+                .add_service(HealthServer::new(HealthService {
+                    mock_server,
+                }))
+                .serve_with_incoming(TcpListenerStream::new(listener))
+                .await;
+        }
+    });
+    MockServer {
+        _server: server,
+        local_addr,
+        mocked: mock_server,
+    }
+}
+
+struct HealthService {
+    mock_server: astria_grpc_mock::MockServer,
+}
+
+#[tonic::async_trait]
+impl Health for HealthService {
```

Comment:
what's the purpose of the health service? is it just for testing the mock server?

Reply:
yeah